ec/provider_config.go

```diff
@@ -33,22 +35,36 @@ const (
 )
 
 // configureAPI implements schema.ConfigureFunc
-func configureAPI(d *schema.ResourceData) (interface{}, error) {
+func configureAPI(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	timeout, err := time.ParseDuration(d.Get("timeout").(string))
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
```

Comment:
This won't be necessary any more if we use the `FromErr` function, and we are not building a collection of several diagnostics